pkg/monitoring/prometheusagent/sharding/sharding.go

```diff
@@ -40,7 +40,10 @@ func (s Strategy) ComputeShards(currentShardCount int, timeSeries float64) int {
 
 	// We always have a minimum of 1 agent, even if there is no worker node
 	if desiredShardCount <= 0 {
-		return 1
+		if currentShardCount <= 0 {
+			return 1
+		}
+		return currentShardCount
```

Comment:
There's no way currentShardCount is <= 0 so.might as well return currentShardCount right ?

Reply:
The default is 1 and it can only increase

Reply:
semantically currentShardCount could be 0, and there is a unit test with this value. So I would rather keep it like this to be safe.

Reply:
Sure but then can you return `math.Max(commonmonitoring.DefaultShards, currentShardCount)` instead?

Reply:
done